src/controllers/fxaRpEvents.js

```diff
@@ -0,0 +1,167 @@
+import mozlog from '../utils/log.js'
+import appConstants from '../appConstants.js'
+const log = mozlog('controllers.fxa-rp-events')
+
+const FXA_PROFILE_CHANGE_EVENT = 'https://schemas.accounts.firefox.com/event/profile-change'
+const FXA_SUBSCRIPTION_CHANGE_EVENT = 'https://schemas.accounts.firefox.com/event/subscription-state-change'
+const FXA_DELETE_USER_EVENT = 'https://schemas.accounts.firefox.com/event/delete-user'
+
+/**
+ * Fetch FxA JWT Public for verification
+ *
+ * @returns {Array} keys an array of FxA JWT keys
+ */
+const getJwtPubKey = async () => {
+  jwtKeyUri = `${appConstants.OAUTH_ACCOUNT_URI}/jwt`
+  try {
+    const res = await fetch(jwtKeyUri, {
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    })
+    const { keys } = res
+    log.info('getJwtPubKey', `fetched jwt public keys from: ${jwtKeyUri} - ${keys}`)
+    return keys
+  } catch (e) {
+    captureException('Could not get JWT public key', jwtKeyUri)
+  }
+}
+
+/**
+ * Authenticate FxA JWT for FxA relay event requests
+ *
+ * @param {*} req
+ * @returns {object} decoded JWT data, which should contain FxA events
+ */
+const authenticateFxaJWT = async (req) => {
+  // Assuming this is how you retrieve your auth header.
+  const authHeader = req?.headers?.authorization
+
+  // Require an auth header
+  if (!authHeader) {
+    captureException('No auth header found', req?.headers)
+    throw UnauthorizedError('No auth header found')
+  }
+
+  // Extract the first portion which should be 'Bearer'
+  const headerType = authHeader.substr(0, authHeader.indexOf(' '))
+  if (headerType !== 'Bearer') {
+    throw UnauthorizedError('Invalid auth type')
+  }
+
+  // The remaining portion, which should be the token
+  const headerToken = authHeader.substr(authHeader.indexOf(' ') + 1)
+
+  // Decode the token, require it to come out ok as an object
+  const token = jwt.decode(headerToken, { complete: true })
+  if (!token || typeof token === 'string') {
+    throw UnauthorizedError('Invalid token type')
+  }
+
+  // Verify we have a key for this kid, this assumes that you have fetched
+  // the publicJwks from FxA and put both them in an Array.
+  const publicJwks = await getJwtPubKey()
+  const jwk = publicJwks.find(j => j.kid === token.header.kid)
+  if (!jwk) {
+    throw UnauthorizedError('No jwk found for this kid: ' + token.header.kid)
+  }
+  const jwkPem = jwkToPem(jwk)
+
+  // Verify the token is valid
+  const decoded = jwt.verify(headerToken, jwkPem, {
+    algorithms: ['RS256']
+  })
+  // This is the JWT data itself.
+  return decoded
+}
+
+/**
+ * Handler for FxA events, used by FxA as a callback URI endpoint
+ * Example events include FxA user deletion, profile changes, and subscription changes
+ *
+ * @param {*} req
+ * @param {*} res
+ * @returns
+ */
+const fxaRpEvents = async (req, res) => {
+  let decodedJWT
+  try {
+    decodedJWT = await authenticateFxaJWT(req)
+  } catch (e) {
+    log.error('fxaRpEvents', e)
+    captureException(e)
```

Comment:
I think this is the proper expected use of `captureException` :) And same below.